discovery/spec.md

```diff
@@ -55,305 +45,194 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+  - MUST be a non-empty URI
+
 - Examples:
-  - `42`
-  - `915148800`
+  - `https://example.com/discovery/endpoints/1234`
 
-##### name
+#### epoch
 
-- Type: `String`
-- Description: A unique human readable identifier for this Service. This value
-  MUST be unique (case insensitive) within the scope of this Discovery Endpoint.
+- Type: Unsigned Integer
+- Description: A number representing the version number of this Endpoint. Each
+  time this Endpoint is modified this property MUST be updated with a new value
+  that is greater than the current one.
 - Constraints:
   - REQUIRED
-  - MUST be a non-empty string
+  - MUST be an unsigned integer
 - Examples:
-  - `my storage service`
-  - `cool git offering`
-
-##### url
-
-- Type: `URL`
-- Description: An absolute URL that references this Service within this
-  Discovery Endpoint. This value MUST be usable in subsequent requests, by
-  authorized clients, to retrieve this Service
-  entity.
-  Note: this value will most likely be Discovery Endpoint defined and therefore
-  will most likely be ignored when provided by clients.
+  - `0`
+  - `1001`
+
+#### description
+
+- Type: String
+- Description: A brief summary of the purpose of the Endpoint.
 - Constraints:
-  - MUST be specified by the Discovery Endpoint and otherwise read-only
-  - REQUIRED in responses from the Discovery Endpoint.
-  - MUST be a non-empty URL
-  - MUST end with `fsegments` (per RFC1738) of: `/services/{id}` where `id` is
-    the Service's `id` attribute.
+  - OPTIONAL
+  - if present, MUST be a non-empty string
 - Examples:
-  - `http://example.com/services/bf5ff5cc-d059-4c79-a89a-2513e45a1340`
-  - `http://discovery.github.com/services/892abefc-0293-9273-bead-92830efaefa0`
+  - "A queue of the sensor generated messages"
 
-##### description (Service)
+#### tags
 
-- Type: `String`
-- Description: Human readable description.
+- Type: Map of name/value string pairs
+- Description: A mechanism in which additional metadata about the Endpoint can
+  be stored.
+
+  This property MAY be serialized as an empty map or MAY excluded from the
+  serialization entirely if there is no value.
 - Constraints:
   - OPTIONAL
-  - If present, MUST be a non-empty string
+  - if present, MUST be a map of name/value string pairs
+  - Each name MUST be a non-empty, unique within the scope of this map, string.
+    Values MAY be empty strings
+- Examples:
+  - `{ "owner": "John", "isVerified: "" }`
 
-##### docsurl
+#### docsurl
 
-- Type: `URL`
-- Description: Absolute URL that provides a link to additional documentation
-  about the service. This is intended for a developer to use in order to learn
-  more about this service's events produced.
+- Type: URL
+- Description: A URL to additional documentation about this Endpoint. This
+  specification does not define any constraints on the data returned from this
+  URL.
 - Constraints:
   - OPTIONAL
-  - If present, MUST be a non-empty absolute URI
+  - if present, MUST be a non-empty URL
```

Comment:
We might need to declare URL is only `HTTP/HTTPS`

Reply:
done